src/chainstate/stacks/db/blocks.rs

```diff
@@ -5226,14 +5226,24 @@ impl StacksChainState {
     }
 
     /// Check to see if a transaction can be (potentially) appended on top of a given chain tip.
-    /// Note that this only checks the transaction against the _anchored chain tip_, not the
-    /// unconfirmed microblock stream trailing off of it.
+    /// The input `mempool_admission_check` determines what the transaction is validated against.
+    /// If `mempool_admission_check`==OffChainOnly, the transaction will only be validated against unconfirmed state.
+    /// If `mempool_admission_check`==OnChainOnly, the transaction will only be validated against confirmed state.
+    /// If `mempool_admission_check`==Any, the transaction will be accepted if it validates against the unconfirmed tip.
```

Comment:
What's the difference between the `Any` and `OffChainOnly` case here?

Reply:
There is no difference between the `Any` and `OffChainOnly` case - I can consolidate the comments. 

I left the enum as is, and now I take it as what the user has specified. `Any` signifies the user has no specific preference, and `OffChainOnly`/`OnChainOnly` signify that the query parameter was set. Let me know if I should instead specify a new enum that omits the `Any` case.

Reply:
I'd very much prefer a separate enum, since this enum here is used for an entirely different purpose that does not represent what the `mempool_admission_check` query parameter is doing.  The `mempool_admission_check` parameter is ultimately specifying a non-consensus-critical admission policy, and we can expect to expand the policy in the future.  The `TransactionAnchorMode` enum, by contrast, is a consensus-critical enum that lists the ways transaction anchor modes are represented and encoded in the system.  These are not even close to the same things in my mind.